src/controllers/scripting/controllerscriptenginebase.cpp

```diff
@@ -0,0 +1,251 @@
+    // use VERIFY_OR_DEBUG_ASSERT here
+    if (!m_pJSEngine) {
+        return false;
+    }
+
+    if (functionObject.isError()) {
+        qDebug() << "ControllerScriptHandlerBase::executeFunction:"
+                 << functionObject.toString();
+        return false;
+    }
+
+    // If it's not a function, we're done.
+    if (!functionObject.isCallable()) {
+        qDebug() << "ControllerScriptHandlerBase::executeFunction:"
+                 << functionObject.toVariant() << "Not a function";
+        return false;
+    }
+
+    // If it does happen to be a function, call it.
+    QJSValue returnValue = functionObject.call(args);
+    if (returnValue.isError()) {
+        showScriptExceptionDialog(returnValue);
+        return false;
+    }
+    return true;
+}
+
+void ControllerScriptEngineBase::showScriptExceptionDialog(
+        QJSValue evaluationResult, bool bFatalError) {
+    VERIFY_OR_DEBUG_ASSERT(evaluationResult.isError()) {
+        return;
+    }
+
+    QString errorMessage = evaluationResult.toString();
+    QString line = evaluationResult.property("lineNumber").toString();
+    QString backtrace = evaluationResult.property("stack").toString();
+    QString filename = evaluationResult.property("fileName").toString();
+
+    QString errorText;
+    if (filename.isEmpty()) {
+        errorText = QString("Uncaught exception at line %1 in passed code.")
+                            .arg(line);
+    } else {
+        errorText = QString("Uncaught exception at line %1 in file %2.")
+                            .arg(line, filename);
+    }
+
+    errorText += QStringLiteral("\n\nException:\n  ") + errorMessage;
+
+    // Do not include backtrace in dialog key because it might contain midi
+    // slider values that will differ most of the time. This would break
+    // the "Ignore" feature of the error dialog.
+    QString key = errorText;
+    qWarning() << "ControllerScriptHandlerBase:" << errorText;
+
+    // Add backtrace to the error details
+    errorText += QStringLiteral("\n\nBacktrace:\n") + backtrace;
+
+    if (!m_bDisplayingExceptionDialog) {
+        scriptErrorDialog(errorText, key, bFatalError);
+    }
+}
+
+void ControllerScriptEngineBase::scriptErrorDialog(
+        const QString& detailedError, const QString& key, bool bFatalError) {
+    if (m_bTesting) {
+        return;
+    }
+
+    ErrorDialogProperties* props =
+            ErrorDialogHandler::instance()->newDialogProperties();
+
+    QString additionalErrorText;
+    if (bFatalError) {
+        additionalErrorText =
+                tr("The functionality provided by this controller mapping will "
+                   "be disabled until the issue has been resolved.");
+    } else {
+        additionalErrorText =
+                tr("You can ignore this error for this session but "
+                   "you may experience erratic behavior.") +
+                QString("<br>") +
+                tr("Try to recover by resetting your controller.");
```

Comment:
Does this make any difference? I think if the controller needs to be reset, it is a fatal error. 
For me this sentence fits more to the fatal error.

Reply:
This happens when an exception is throws in an input handler (e. g. when pressing a button on the midi controller). You have the option to ignore the same error and continue your set. The button might not work if there's a bug in the mapping, but the other buttons probably will.

If this happens during a gig, you usually don't want to shut down mixxx and look through the mapping's JS code to fix the issue, just don't use that button anymore.

Reply:
Please add the info from @Holzhaus as a source code comment.

Reply:
This is already clear from the name of `bFatalError`. IMO there is no point adding a comment repeating that.

Reply:
No, Not the details @Holzhaus has explained.

Reply:
I am not adding a code comment here. The code is clear. If you want to change the wording of the error dialog, that is beyond the scope of this PR and I will not discuss it here.

Reply:
What is the reason for rejecting this requests? 

The task of the reviewer is t o check if the code if understandable. I did not understand it in the first place. We have spend some time to make it clear. It is quite natural to save this work for others as a source code comment.

Reply:
I have reread the code multiple times and the code is clear. @Holzhaus' explanation above is just repeating what the code already says. If the error is not fatal, let the user continue. Comments repeating the code are noise.

Reply:
Ping! My initial confident has proved that the comment is missing.
This is one main topic in a review. Check if any code is understandable.

Reply:
The proposed comment is waiting for integration or a respond for 6 weeks now. Please integrate of rephrase to make progress.